### PR TITLE
[DQM/L1TEMU] Removing the condition for data/emul sizes being equal  - backport

### PR DESCRIPTION
**Pull request by vukasinmilosevic**
Removing the condition for data/emul sizes being equal in order to fill problematic sums histograms.
Backport of PR #25087

### CODE REVIEW

**Comment by andrius-k** [comment]
please test

**Comment by andrius-k** [comment]
please test

File: DQM/L1TMonitor/src/L1TdeStage2CaloLayer2.cc [hunk 1]
```diff
@@ -1012,568 +1012,571 @@ bool L1TdeStage2CaloLayer2::compareSums(
   double dataPhi = 0;
   double emulPhi = 0;
 
-  l1t::EtSumBxCollection::const_iterator dataIt = dataCol->begin(currBx);
+  l1t::EtSumBxCollection::const_iterator dataIt;
   l1t::EtSumBxCollection::const_iterator emulIt = emulCol->begin(currBx);
 
   // if either data or emulator collections are empty, or they have different
   // size, mark the event as bad (this should never occur in normal running)
```
Comment (thomreis) [inline DQM/L1TMonitor/src/L1TdeStage2CaloLayer2.cc]: Hi @vukasinmilosevic 
can you please also update the comment corresponding to the changed code?

**Comment by andrius-k** [comment]
Hi @vukasinmilosevic, are you planning to test this PR in playback?

File: DQM/L1TMonitor/src/L1TdeStage2CaloLayer2.cc (hunk 1 above)
Comment (vukasinmilosevic) [reply DQM/L1TMonitor/src/L1TdeStage2CaloLayer2.cc]: Ah yes. Will do it now.

**Comment by vukasinmilosevic** [comment]
Hi @andrius-k. Yes, I will send the official request for it to be deployed as soon as I address the comment from @thomreis.

**Comment by andrius-k** [comment]
@vukasinmilosevic good, take your time.

**Comment by andrius-k** [comment]
please test

**Comment by vukasinmilosevic** [comment]
Hello @andrius-k, could you wait with merging this PR (as well as the original)? The dqm playback testing is returning different result than my local testing.

**Comment by andrius-k** [comment]
Hi @vukasinmilosevic, sure, it can wait.

**Comment by andrius-k** [comment]
Hi @vukasinmilosevic, the indentation of your recent changes seems very inconsistent. Could you please take a look at it?

**Comment by thomreis** [comment]
Backport of PR #25087

**Comment by thomreis** [comment]
Backport of #25087

**Comment by thomreis** [comment]
please test

**Comment by andrius-k** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**